src/linpde_gp/randprocs/covfuncs/_tensor_product.py

```diff
@@ -61,6 +61,26 @@ def _keops_lazy_tensor(
             operator.mul, lazy_tensor_dimensionwise(self._factors, x0s, x1s)
         )
 
+    def linop(
+        self, x0: ArrayLike, x1: Optional[ArrayLike] = None
+    ) -> pn.linops.LinearOperator:
+        # Use Kronecker-based linop if possible
+        if isinstance(x0, TensorProductGrid) and (
+            x1 is None or isinstance(x1, TensorProductGrid)
+        ):
+            if isinstance(x1, TensorProductGrid):
+                x1_factors = x1.factors
+            else:
+                x1_factors = [None for _ in x0.factors]
+            kronecker_factors = [
+                self._factors[idx].linop(x0_factor, x1_factor)
+                for (idx, (x0_factor, x1_factor)) in enumerate(
+                    zip(x0.factors, x1_factors)
+                )
+            ]
```

Comment:
The cases in which only one of `x0` and `x1` are `TensorProductGrid`s should also result in a `Kronecker` structured matrix. Do you think we could handle this case here too? Seems like this might even simplify the code